pkg/instrumentation/sdk.go

```diff
@@ -56,39 +56,48 @@ type sdkInjector struct {
 	client client.Client
 }
 
-func (i *sdkInjector) inject(ctx context.Context, insts languageInstrumentations, ns corev1.Namespace, pod corev1.Pod) corev1.Pod {
+func (i *sdkInjector) inject(ctx context.Context, insts languageInstrumentations, ns corev1.Namespace, pod corev1.Pod, containerName string) corev1.Pod {
```

Comment:
`chooseServiceName(pod corev1.Pod, resources map[string]string) string {` 

has to be updated as well to correctly choose the service name.


`createResourceMap(ctx context.Context, otelinst v1alpha1.Instrumentation, ns corev1.Namespace, pod corev1.Pod) map[string]string {`

has to be as well updated to choose the right container.

Reply:
Ok thank you for your feeback. I Will take a closer on 21 Feb week and also write one kuttl test.